robottelo/cli/base.py

```diff
@@ -374,6 +374,13 @@ def _construct_command(cls, options=None):
                 if val is True:
                     tail += u' --{0}'.format(key)
                 elif val is not False:
+                    if isinstance(val, list):
+                        val = ','.join(
+                            str(el.get('id') or el.get('name'))
```

Comment:
Case 1 is OK.

Case 2 is better to let fail than to do some "magic". As @Ichimonji10 said if the user is lazy and a uuid is needed instead of a name getting the id or name key will not work. With that said is better to not have a special case and deal everything the same way, in other words the user should pass a list comprehension.